UPGRADING.md

```diff
@@ -1,3 +1,87 @@
+* [Configuration changes](#configuration-changes)
+* [Command line changes](#command-line-changes)
+
+# Configuration changes
+
+This document covers most of the breaking changes in using the command line or configuring the different softwares.
+
+We also recommend that node operator generates a bare configuration (`vega init...`, `vega datanode init...`) in a temporary folder to compare this new configuration to their existing ones to address any changes. Any missing entries in the configuration will be using defaults.
+
+## Vega
+
+The Vega configuration file can be found in `$VEGA_HOME/config/node/config.toml`.
+
+## Data node
+
+The data node configuration file can be found in `$DATANODE_HOME/config/data-node/config.toml`.
+
+### Settings added in 0.71.4
+
+**_Rate Limiting_** - In the past the datanode API did rate limiting only by the number of subscriptions open per IPs. With the new version a rate limit per IP on the unary calls for the REST, GraphQL and grpc APIs was introduced. The server now allows 20 API calls per second on average with a burst of up to 100 api calls in a short period of time.
+For details how to use this setting and configure this functionality, please refer to [rateLimiting](https://github.com/vegaprotocol/vega/blob/develop/datanode/ratelimit/README.md)
+
+**_REST and GraphQL configuration_** - Previously the configuration of the Graphql and REST APIs used separate HTTP servers and configuration. This was merge onto a single server now. Also the default endpoint for Graphql changed from /query to /graphql
+Usage example:
+```Toml
+[Gateway]
+  Port = 3008
+  IP = "0.0.0.0"
+  [Gateway.GraphQL]
+    Enabled = true
+    Endpoint = "/graphql"
+  [Gateway.REST]
+    Enabled = true
+```
+
+**_IPFS update_** - The ipfs library used by the datanode to share network history have been updated. This may require manual steps to reset the state and apply the ipfs migrations.
+
+### Settings removed in 0.71.4
+
+**_X-Vega-Connection_** - This HTTP header is now deprecated from API calls to the datanode.
+
+## Visor
+
+### Breaking changes with 0.71.4
+
+Visor used to require both the vega and datanode binaries. With recent changes, the vega binary is now embedding the datanode cli as a subcommands. This has been reflected to the visor configuration as now a single binary is required in the configuration.
+
+Here's an example of visor configuration reflecting these changes:
+```Toml
+maxNumberOfFirstConnectionRetries = 43200
+maxNumberOfRestarts = 3
+restartsDelaySeconds = 5
+stopDelaySeconds = 3
+stopSignalTimeoutSeconds = 15
+
+[upgradeFolders]
+  "vX.X.X" = "vX.X.X"
+
+[autoInstall]
+  enabled = true
+  repositoryOwner = "vegaprotocol"
+  repository = "vega"
+  [autoInstall.assets]
+    assset_name = "vega-linux-amd64.zip"
+    binary_name = "vega"
+```
+
+# Command line changes
+
+## Vega
+
+#### Deprecations
+
+`IssueSignatures` is no longer a validator command and is now protected by the spam engine.
```

Comment:
`spam engine` may be better phrased as "spam protection" or "spam protection engine"? for the sake of consistency, as I don't think it's called the spam engine anywhere I've seen it described.